### ISSUE TITLE
[docs] no-direct-set-state-in-use-effect question

### ISSUE THREAD

**Issue by Stanzilla** [Rel1cx/eslint-react#663]
When looking at code like 

```ts
useEffect(() => {
    setShowButton(!isLoading && (!user || (user && !user.hideAds)));
  }, [isLoading, user]);
```

I correctly get a warning. When I then change it to something like 

```ts
const updateShowButton = useCallback(() => {
    setShowButton(!isLoading && (!user || (user && !user.hideAds)));
}

useEffect(() => {
    updateShowButton();
  }, [updateShowButton]);
```

I no longer get the warning. Now, the question is, since this basically just hides the `set` call, should this also be a violation? If yes, this is a bug, if not, I think the docs should be updated to reflect that.

**Comment by Rel1cx** [comment]
Sorry for the late reply, this is indeed a violation, I'll be working on a fix for this soon!
